primitives/parachain-inherent/src/client_side.rs

```diff
@@ -18,23 +18,32 @@
 
 use crate::ParachainInherentData;
 use codec::Decode;
+
 use cumulus_primitives_core::{
-	relay_chain::{self, v2::HrmpChannelId, Hash as PHash},
-	ParaId, PersistedValidationData,
+	relay_chain::{self, v2::HrmpChannelId, DmqContentsBounds, Hash as PHash},
+	DmpMessageWindow, DmpQueueState, ParaId, PersistedValidationData,
 };
+
 use cumulus_relay_chain_interface::RelayChainInterface;
+use relay_chain::well_known_keys as relay_well_known_keys;
 
 const LOG_TARGET: &str = "parachain-inherent";
+// A hard limit for the amount of space used by the messages we pass in `ParachainInherentData`.
+// This is probably good to promote as configuration parameter.
+// TODO(optimization): remove the messages that we've not touched from the storage proof.
+const MESSAGE_PROCESSING_CAPACITY: usize = 1 * 1024 * 1024; // 1MB
+															// How many dmp queue pages we fetch at once. Page size is defined by the relay chain dmp storage.
```

Comment:
nit: what happened here with indentation? :)

Reply:
lol, it has a mind of its own :P